src/github.com/matrix-org/dendrite/roomserver/input/latest_events.go

```diff
@@ -0,0 +1,72 @@
+package input
+
+import (
+	"bytes"
+	"github.com/matrix-org/dendrite/roomserver/types"
+	"github.com/matrix-org/gomatrixserverlib"
+)
+
+// updateLatestEvents updates the list of latest events for this room.
+func updateLatestEvents(
+	db RoomEventDatabase, roomNID types.RoomNID, stateAtEvent types.StateAtEvent, event gomatrixserverlib.Event,
+) error {
+	oldLatest, updater, err := db.GetLatestEventsForUpdate(roomNID)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			// Commit if there wasn't an error.
+			// Set the returned err value if we encounter an error committing.
+			err = updater.Close(true)
+		} else {
+			// Ignore any error we get rolling back since we don't want to
+			// clobber the current error
+			updater.Close(false)
```

Comment:
What's the worst thing that can happen if we fail to rollback? Memory leak? Connection leak? I'm wary about dropping this error to the floor if it could indicate a real problem.